sasmodels/models/micromagnetic_FF_3D.py

```diff
@@ -0,0 +1,194 @@
+scattering function of the longitudinal magnetisation $S_M$ reflects the
+variations of the saturation magnetisation, e.g. jumps at the particle-matrix
+interface. $R_K$ and $R_M$ denote the micromagnetic response functions that
+describe the magnetisation distribution around a perturbation in magnetic
+anisotropy and flucutations in the saturation magnetisation value.
+
+.. figure:: img/micromagnetic_FF.png
+
+    Magnetisation distribution around (left) a particle with magnetic easy axis
+    in the vertical direction and (right) a precipitation with a magnetisation
+    that is higher than the matrix phase.  
+
+The micromagnetic response functions depend on magnetic material parameters $M_S$:
+average saturation magnetisation of the material, $H_i$: the internal magnetic
+field, $A$ the average exchange-stiffness constant. In the vicinity of lattice 
+imperfection in ferromagnetic materials,  antisymmetric Dzyaloshinskii–Moriya
+interaction (DMI) can occur due to the local structural inversion symmetry
+breaking [#Arrott1963]_. DMI with strength $D$ can give rise to nonuniform spin
+textures resulting in a polarization-dependent asymmetric scattering term for 
+polycrystalline ferromagnetic with a centrosymmetric crystal structure [#Michels2016]_.
+We assume (for simplicity) an isotropic microstructure (for $S_M$) and random 
+orientation of magnetic easy axes (additionally for $S_K$) such that the 
+contributions of the magnetic microstructure only depend on the magnitude of $q$.
+Considerations for a microstructure with a prefered orientation (texture) can be
+found in [#Weissmueller2001]_. In the code the averaging procedure over the random
+anisotropy is explicitely performed. A specific orientation distribution can be
+implemented. 
+
+The magnetic field is oriented with an inclination of $alpha$ to the neutron beam
+and rotated by $beta$. The model for the nuclear scattering amplitude, saturation
+magnetisation is based on spherical particles with a core shell structure. For
+simplicity, only the core has an effective anisotropy, that is varying randomly
+in direction from particle to particle. The effect of different, more complex 
+spatial profiles of the anisotropy can be seen in Michels PRB 82, 024433 (2010).
+The magnetic scattering length density (SLD) is defined as 
+$
ho_{mathrm{mag}}=b_H M_S$, where $b_H= 2.91*10^{8}A^{-1}m^{-1}$ and $M_S$
+is the saturation magnetisation (in $A/m$).
+
+The fraction of "upward" neutrons before ('up_frac_i') and after the sample 
+('up_frac_f') must range between 0 to 1, with 0.5 denoting an unpolarised beam.
+Note that a fit may result in a negative magnetic SLD, and hence magnetisation, 
+when the polarisation state is inverted, i.e. if you have analysed for a $I_{00}$
+ state wheras your data are $I_{11}$. The model allows to construct the 4 
+ spin-resolved cross sections (non-spin-flip I_{00}, I_{11} and spin-flip, here
```

Comment:
All of a sudden a random space seems to be added to the start of every line after this. This is also causing problems for CI with an `unexpected indent` warning. Please remove these.

Reply:
It looks like a search and replace went wrong and a bunch of latex maths disappeared like \times, \who. I'll go over the model description again.